tests/components/wemo/__init__.py

```diff
@@ -0,0 +1,78 @@
+"""The tests for WeMo component."""
+
+from homeassistant.components.wemo import WEMO_MODEL_DISPATCH
+from homeassistant.components.wemo.const import DOMAIN
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+
+class DeviceTests:
+    """Basic name/id/info tests for all wemo device entities."""
+
+    DEVICE_MODEL = "WeMoDevice"
+    DEVICE_SERIAL_NUMBER = "wemodevice_serialnumber"
+    DEVICE_NAME = "wemodevice_name"
+
+    def test_name(self, wemo_device):
+        """Verify the device name."""
+        assert wemo_device.name == self.DEVICE_NAME
+
+    def test_unique_id(self, wemo_device):
+        """Verify the unique id."""
+        assert wemo_device.unique_id == self.DEVICE_SERIAL_NUMBER
+
+    def test_device_info(self, wemo_device):
+        """Verify the device info."""
+        assert wemo_device.device_info == {
+            "name": self.DEVICE_NAME,
+            "identifiers": {(DOMAIN, self.DEVICE_SERIAL_NUMBER)},
+            "model": self.DEVICE_MODEL,
+            "manufacturer": "Belkin",
+        }
+
+
+class SubscriptionTests:
+    """Test that push subscriptions are working."""
+
+    async def test_update_basic_state(self, hass, wemo_device, pywemo_registry):
+        """A subscription callback will update the device _state."""
+        hass.data.setdefault(DOMAIN, {}).setdefault("registry", pywemo_registry)
```

Comment:
Please don't interact with any integration details in the tests. Set up the integration and assert core state. That's either the state machine, the entity registry, the device registry or the config entry. Patch in the periphery as needed, usually that means the device library.

We want to write the tests like that to make them more robust so we can refactor the integration without changing the tests.